### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by greenemcg**
Ready for editor building to go on further.....

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -75,30 +50,11 @@ typings/
...
-
-# DynamoDB Local files
-.dynamodb/
+node_modules
 
-# TernJS port file
-.tern-port
+lib
+.idea
+/tests/*.test.js
```
Comment (AndrewBird81) [inline .gitignore:57]: Why are we ignoring test.js files?

File: src/antlr/CqlAntlrListener.ts [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+import CqlCodeSystem from "./dao/CqlCodeSystem";
+import CqlValueSet from "./dao/CqlValueSet";
+import CqlCode from "./dao/CqlCode";
+import LineInfo from "./dao/LineInfo";
+import {ParserRuleContext} from "antlr4ts/ParserRuleContext";
+import {Token} from "antlr4ts/Token";
+import {ParseTree} from "antlr4ts/tree";
+
+
+abstract class CreatorBase<T extends CqlText> {
```
Comment (AndrewBird81) [inline src/antlr/CqlAntlrListener.ts:26]: I think we should have one class per file

**Review by AndrewBird81 — CHANGES_REQUESTED** [review]

File: .github/workflows/npm-publish.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  npm-publish:
+    name: npm-publish
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v2-beta
```
Comment (adongare) [inline .github/workflows/npm-publish.yml:16]: setup-node@v2 is available now

File: .github/workflows/unit_test_coverage.yml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      - name: Download coverage artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: coverage
+
+      - name: Upload code coverage to Codacy
+        uses: codacy/codacy-coverage-reporter-action@master
+        with:
+          project-token: ${{ secrets.CODACY_PROJECT_TOKEN }}
+          coverage-reports: lcov.info
```
Comment (adongare) [inline .github/workflows/unit_test_coverage.yml:86]: Don't think we need this.

**Review by adongare — APPROVED** [review]

**Review by AndrewBird81 — APPROVED** [review]

**Pull request merged**